Sources/Harmony/Data/Query/NetworkQuery.swift

```diff
@@ -0,0 +1,163 @@
+// limitations under the License.
+//
+
+import Foundation
+import Alamofire
+
+open class NetworkQuery: KeyQuery {
+
+    public enum Method: Equatable {
+        
+        case get
+        case delete
+        case put(type: ContentType?)
+        case post(type: ContentType?)
+
+        public static func ==(lhs: NetworkQuery.Method, rhs: NetworkQuery.Method) -> Bool {
+            switch (lhs, rhs) {
+            case (.get, .get):
+                return true
+            case(.delete, .delete):
+                return true
+            case (.put, .put):
+                return true
+            case (.post, .post):
+                return true
+            default:
+                return false
+            }
+        }
+        
+        public func with(contentType newContentType: ContentType?) -> Self {
+            if case .put = self {
+                return .put(type: newContentType)
+            } else if case .post = self {
+                return .post(type: newContentType)
+            } else {
+                return self
+            }
+        }
+        
+        public func contentType() -> NetworkQuery.ContentType? {
+            switch self {
+            case .get:
+                return nil
+            case .delete:
+                return nil
+            case .put(type: let type):
+                return type
+            case .post(type: let type):
+                return type
+            }
+        }
+    }
+
+    public enum ContentType {
+        case FormUrlEncoded(params: [String: String])
+        case Json(entity: Encodable)
+    }
+
+    private let path: String
+    private let params: [String: Any]
+    private let headers: [String: String]
+    public var method: Method
+    public let key: String
+    
+    public init(method: Method, path: String, params: [String : Any] = [:], headers: [String: String] = [:], key: String? = nil) {
+        self.method = method
+        self.path = path
+        self.params = params
+        self.headers = headers
+
+        self.key = key ?? "\(path) ? \(params.map { key, value -> String in "\(key) = \(value)" }.joined(separator: "&"))"
+    }
+}
+
+extension NetworkQuery {
+    
+    private func mapToAlamofireMethod(method: Method) -> HTTPMethod {
+        switch method {
+        case .get:
+            return .get
+        case .delete:
+            return .delete
+        case .put:
+            return .put
+        case .post:
+            return .post            
+        }
+    }
+
+    public func request(url: URL, session: Session) -> DataRequest {
+        let fullUrl = url.appendingPathExtension(path)
+        let afMethod = mapToAlamofireMethod(method: method)
+        var parameters: [String: Any] = self.params
+        let encoding: URLEncoding = .default
+        let headers = HTTPHeaders(self.headers)                
+        
+        if case let .put(type: contentType) = method {
+            if let contentType {
```

Comment:
In this line I am having a compiler error

Reply:
@franmontiel, all working from my side. Did you use the latest XCode version? (14.0.1)

Reply:
That's right, you need XCode 14, since I am using some features from Swift 5.7